From e4c96cb82b595e813ddc3672579082bc1ae50bf6 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:27:17 -0400
Subject: [tools] perf/list: Add usage

Message-id: <1408462094-14194-269-git-send-email-jolsa@redhat.com>
Patchwork-id: 88304
O-Subject: [PATCH RHEL7.1 BZ1131394 268/325] perf list: Add usage
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit 44d742e01e6d3dd544ee1873b660a3c8bc1413bb
Author: David Ahern <dsahern@gmail.com>
Date: Wed Oct 30 10:28:29 2013 -0600

description
===========
Currently 'perf list' is not very helpful if you forget the syntax:

  $ perf list -h

  List of pre-defined events (to be used in -e):

After:
  $ perf list -h

   usage: perf list [hw|sw|cache|tracepoint|pmu|event_glob]
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/builtin-list.c b/tools/perf/builtin-list.c
index 45000e7..011195e 100644
--- a/tools/perf/builtin-list.c
+++ b/tools/perf/builtin-list.c
@@ -14,20 +14,31 @@
 #include "util/parse-events.h"
 #include "util/cache.h"
 #include "util/pmu.h"
+#include "util/parse-options.h"
 
 int cmd_list(int argc, const char **argv, const char *prefix __maybe_unused)
 {
  int i;
+ const struct option list_options[] = {
+  OPT_END()
+ };
+ const char * const list_usage[] = {
+  "perf list [hw|sw|cache|tracepoint|pmu|event_glob]",
+  NULL
+ };
+
+ argc = parse_options(argc, argv, list_options, list_usage,
+        PARSE_OPT_STOP_AT_NON_OPTION);
 
  setup_pager();
 
- if (argc == 1) {
+ if (argc == 0) {
   print_events(NULL, false);
   return 0;
  }
 
- for (i = 1; i < argc; ++i) {
-  if (i > 2)
+ for (i = 0; i < argc; ++i) {
+  if (i)
    putchar('\n');
   if (strncmp(argv[i], "tracepoint", 10) == 0)
    print_tracepoint_events(NULL, NULL, false);
-- 
1.7.1